### PR TITLE
Feat/UI/reservation

### PR DESCRIPTION
**Pull request by sammy-code98**
### **Pull Request Summary**

This pull request introduces the following changes to the reservation feature:

#### **Features**
1. **Reservation Page and Routes Setup** (`ae0ec1d`)
   - Implemented the foundational setup for the reservation page and its corresponding routes.
   
2. **Reservation Cards** (`5d7054d`)
   - Added reservation cards categorized as:
     - Incoming
     - Confirmed
     - Rejected

3. **Accept Reservation Modal** (`3c3b5d5`)
   - Implemented a modal for accepting reservations.

4. **Reject Reservation Modal** (`ea5524d`)
   - Added a modal for rejecting reservations.

#### **Data and Type Additions**
1. **Incoming Reservations** (`50d6d2a`)
   - Defined types and added dummy data for incoming reservations.
   
2. **Confirmed Reservations** (`4c00c7f`)
   - Defined types and added dummy data for confirmed reservations.

3. **Rejected Reservations** (`e556496`)
   - Added dummy data for rejected reservations.

#### **Refactor**
1. **Transition from Cards to Tables** (`a6943d7`)
   - Refactored the reservation components to use tables instead of cards for improved clarity and structure.

#### **Fixes**
1. **Default Value Issue** (`ea5524d`)
   - Resolved an issue where the `reserveNo` field had incorrect default values.

---

### **Additional Notes**
- The implementation uses dummy data to simulate the reservation flow and will be updated with actual API data in the future.
- Modals for accepting and rejecting reservations provide a user-friendly interface for managing reservation statuses.
- Transitioning to tables ensures scalability and better readability for reservation data. 

This PR is currently marked as **Draft** and open for further review and testing.

### CODE REVIEW

File: host/src/types/routes.ts [hunk 1]
```diff
@@ -14,6 +14,7 @@ export const DashboardRoutes = {
   settings: "/settings",
   profile: "/profile",
   notification: "/notification",
```
Comment (chimdie) [inline host/src/types/routes.ts:16]:
```suggestion
  notifications: "/notifications",
```

File: host/src/types/routes.ts [hunk 2]
```diff
@@ -14,6 +14,7 @@ export const DashboardRoutes = {
...
+  reservation: "/reservation",
```
Comment (chimdie) [inline host/src/types/routes.ts:17]:
```suggestion
  reservations: "/reservations",
```

File: host/src/types/routes.ts
Comment (chimdie) [inline host/src/types/routes.ts:1]: Routes that's shows a list should be plural

File: host/src/pages/Dashboard/Reservation/index.tsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+            </Tab>
+
+            <Tab key="confirm" title="Confirmed Reservation">
+              <div className="p-2">
+                <ConfirmedReservation />
+              </div>
+            </Tab>
+
+            <Tab key="reject" title="Rejected Reservation">
+              <div className="p-2">
```
Comment (chimdie) [inline host/src/pages/Dashboard/Reservation/index.tsx:43]:
```suggestion
            <RejectedReservation />
```

File: host/src/pages/Dashboard/Reservation/index.tsx [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+            variant="underlined"
+          >
+            <Tab key="incoming" title="Incoming Reservations">
+              <div className="py-2">
+                <IncomingReservation />
+              </div>
+            </Tab>
+
+            <Tab key="confirm" title="Confirmed Reservation">
+              <div className="p-2">
```
Comment (chimdie) [inline host/src/pages/Dashboard/Reservation/index.tsx:37]:
```suggestion
              <ConfirmedReservation />
```

File: host/src/pages/Dashboard/Reservation/index.tsx [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+            classNames={{
+              tabList: "gap-12 w-full relative rounded-none p-0 border-b border-divider",
+              cursor: "w-full bg-shorttiee_primary",
+              tab: "max-w-fit px-0 h-12",
+              tabContent: "group-data-[selected=true]:text-black text-grey_400 font-bold",
+            }}
+            variant="underlined"
+          >
+            <Tab key="incoming" title="Incoming Reservations">
+              <div className="py-2">
```
Comment (chimdie) [inline host/src/pages/Dashboard/Reservation/index.tsx:31]:
```suggestion
              <IncomingReservation />
```

File: host/src/pages/Dashboard/Reservation/index.tsx [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+import { Link } from "react-router-dom";
+import { DashboardRoutes } from "@/types/routes";
+import IncomingReservation from "@/components/Reservations/incoming-reservation";
+import ConfirmedReservation from "@/components/Reservations/confirmed-reservation";
+import RejectedReservation from "@/components/Reservations/rejected-reservation";
+
+export default function Reservation(): JSX.Element {
+  return (
+    <div className="py-6 space-y-6">
+      <Breadcrumbs>
```
Comment (chimdie) [inline host/src/pages/Dashboard/Reservation/index.tsx:11]: Do we need still need Breadcrumbs when the Home route is just by the sidebar and `/reservations` route is not a nested route

**Review by chimdie — CHANGES_REQUESTED** [review]

File: host/src/types/routes.ts
Comment (sammy-code98) [reply host/src/types/routes.ts:1]: i will take that under advisement sir

File: host/src/pages/Dashboard/Reservation/index.tsx (hunk 6 above)
Comment (sammy-code98) [reply host/src/pages/Dashboard/Reservation/index.tsx:11]: ohh i think that wont be necessary, would just take it off then

File: host/src/types/routes.ts [hunk 7]
```diff
@@ -8,12 +8,13 @@ export const AuthRoutes = {
 
 export const DashboardRoutes = {
   home: "/dashboard",
-  shortlets: "/shortlet",
+  shortlets: "/shortlets",
   shortletDetail: "/shortlet/:id",
```
Comment (chimdie) [inline host/src/types/routes.ts:12]:
```suggestion
  shortletDetail: "/shortlets/:id",
```

File: host/src/types/routes.ts [hunk 8]
```diff
@@ -8,12 +8,13 @@ export const AuthRoutes = {
...
   editShortlet: "/shortlet/edit/:id",
```
Comment (chimdie) [inline host/src/types/routes.ts:13]:
```suggestion
  editShortlet: "/shortlets/edit/:id",
```

**Review by chimdie — CHANGES_REQUESTED** [review]

File: host/src/types/routes.ts (hunk 8 above)
Comment (micaiah-effiong) [reply host/src/types/routes.ts:13]: I would suggest

```suggestion
  editShortlet: "/shortlet/:id/edit",
```

since it matches to flow of the URI

**Review by chimdie — APPROVED** [review]

**Pull request merged**